service/league_service/league_rater.py

```diff
@@ -18,4 +18,68 @@ def rate(
         outcome: GameOutcome,
         player_rating: Rating,
     ):
+        boost = 0
+        reduction = 0
+        rating = player_rating[0] - 3 * player_rating[1]
+        player_div = cls._get_player_division(current_score.division_id, league)
+        # Highest divison has lowest id
+        lowest_id, highest_id = cls._get_division_id_range()
+
+        if current_score.game_count >= 10:
+            if current_score.division_id is None:
+                return cls._do_placement(league, current_score, rating)
+
+            if rating > player_div.max_rating:
+                boost = 1
+            elif rating < player_div.min_rating:
+                reduction = 1
+            # Linear interpolation of score and rating to have players in top division sorted by rating
+            elif player_div.id == lowest_id and current_score.score < (player_div.highest_score - player_div.lowest_score) \
+                    * (rating - player_div.min_rating) / (player_div.max_rating - player_div.min_rating):
+                boost = 1
+
+            if outcome == "VICTORY":
+                current_score.score += 1 + boost
```

Comment:
I don't think the `+=` syntax works on tuples, given that they are immutable:
```
$ python3
Python 3.6.9 (default, Apr 18 2020, 01:56:04) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import NamedTuple
>>> class A(NamedTuple):
...   a: int
... 
>>> x = A(42)
>>> x.a += 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: can't set attribute
```
You will have to gather the individual entries (`new_division_id`, `new_score`, `game_count`) and then create a new tuple like
```python
return LeagueScore(
  new_division_id,
  new_score,
  current_score.game_count + 1
)
```

Reply:
Done